### ISSUE TITLE
Add check is_callable

### ISSUE THREAD

**Issue by ligurio** [tarantool/checks#35]
It would be convenient to check whether object is callable or not.

We use such check at least in CRUD and DDL and both projects contains it's own implementation:

- https://github.com/tarantool/ddl/blob/master/ddl/check.lua#L684-L713
- https://github.com/tarantool/crud/pull/239

**Comment by Totktonada** [comment]
It would be nice to have such function right in tarantool (in LuaJIT, in fact), because fully correct implementation requires to touch LuaJIT internals. See https://github.com/tarantool/tarantool/issues/3915.

I would also highlight that `luaL_iscallable()` handles cdata, but not userdata.

**Comment by Totktonada** [comment]
I would consider it as blocked on https://github.com/tarantool/tarantool/issues/6816.

**Comment by filonenko-mikhail** [comment]
Sooo, the better thing for developer who wants to make code types safe is to learn static typing languages.

**Comment by Totktonada** [comment]
The dispute about strictness vs flexibility is quite usual: there is very similar topic about database schema strictness. As I see, the world now prefers to have a choice. It allows to reduce effort of transition from prototyping to production code.

**Comment by filonenko-mikhail** [comment]
Such thing have to be done as typescript and js with toolchains, ide supporting. I am also does not feel any customer requests. Use your own custom argument predicate. Much gently closing.
